docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,743 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using `Auth0` OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+With Quarkus, you can easily configure OIDC providers such as link:https://www.keycloak.org/documentation[Keycloak], link:https://developer.okta.com/[Okta], link:https://auth0.com/docs/[Auth0], and other xref:security-openid-connect-providers.adoc[well-known social OIDC and OAuth2 providers].
+
+Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.
+
+== Create Auth0 application
+
+Start with creating an `Auth0` application in the `Auth0` dashboard, for example, lets create a `QuarkusAuth0` web application:
+
+image::auth0-create-application.png[Create Auth0 application]
+
+`Auth0` will create it and generate the client id and secret and show its HTTPS-based domain:
+
+image::auth0-created-application.png[Created Auth0 application]
+
+You will use these 3 properties to configure Quarkus at the next step.
+
+Next, add users to this application by selecting the `Users` option in the application's dashboard, for example:
+
+image::auth0-add-user.png[Add Auth0 application users]
+
+Now that the initial `Auth0` application setup has been completed, you are ready to start creating and configuring a Quarkus endpoint. We will be updating the `Auth0` application configuration along the way as necessary.
+
+== Create Quarkus application
+
+Use the following Maven command to create a Quarkus REST application which can be secured with its OIDC extension.
+
+:create-app-artifact-id: quarkus-auth0
+:create-app-extensions: resteasy-reactive,oidc
+include::{includes}/devtools/create-app.adoc[]
+
+Create the application workspace and import it into your favourite IDE, note a `GreetingResource` JAX-RS resource that will need to be secured:
```

Comment:
US spelling.
- Consider rephrasing this sentence. I'm not quite sure what the note is about here.
- Consider using the admonition markup for notes.

```suggestion
Create the application workspace and import it into your favorite IDE.
Note: <rephrase the`GreetingResource` JAX-RS resource sentence>
```

Reply:
@michelle-purcell I did not mean it to be a `Note`, just wanted to let users know that `GreetingResource` was generated,  which I guess does not warrant a Note-level highlight, can we just replace `note` with some other verb/expression ?

Reply:
@sberyozkin - I see :+1:  In that case, something like...

```suggestion
Create the application workspace and import it into your favorite IDE.
A `GreetingResource` JAX-RS resource gets generated, which you will need to secure.
```

Not sure if my suggestion here ^^ is technically correct. These questions might help to enhance it further:

- Which step in this procedure section actually generates a `GreetingResource` ?
- When do you need to secure it? 
- Will the reader know how to secure it? (or is it covered in the code sample that follows?)

Thanks

Reply:
@michelle-purcell Sounds good, yeah, then we'll work in the tutorial to get it secured